hack/tools/logcheck/main.go

```diff
@@ -27,15 +28,24 @@ import (
 	"golang.org/x/tools/go/analysis/singlechecker"
 )
 
-// Doc explaining the tool.
-const Doc = "Tool to check use of unstructured logging patterns."
+const (
+	// Doc explaining the tool.
+	Doc = "Tool to check use of unstructured logging patterns."
+	// Help text for allow-unstructured flag
+	allowUnstructuredFlagHelpText = `suppress error reports when unstructured logging pattern is used`
+)
 
-// Analyzer runs static analysis.
-var Analyzer = &analysis.Analyzer{
-	Name: "logcheck",
-	Doc:  Doc,
-	Run:  run,
-}
+var (
+	logcheckFlags     = flag.NewFlagSet("", flag.ExitOnError)
+	allowUnstructured = logcheckFlags.Bool("allow-unstructured", false, allowUnstructuredFlagHelpText)
```

Comment:
Please avoid using global variables.

Reply:
@serathius I tried finding a way to do this, but it's either really tricky or not possible.

For flags, I had to use global variable as I couldn't find a way to pass it into `run()` where it's actually used.

For Analyzer, I had to make it global so that anyone can access it from outside. In our case, unit tests can't access it if it's not global and fails.

If you have suggestions on how to do it, I can add those changes to the PR.

Reply:
Yea, it's a little harder then I thought at the beginning.

Try this:
```
type config struct {
	allowUnstructured bool
}

func main() {
	singlechecker.Main(analyser())
}

func analyser() *analysis.Analyzer {
	c := config{}
	logcheckFlags     := flag.NewFlagSet("", flag.ExitOnError)
	logcheckFlags.BoolVar(&c.allowUnstructured, "allow-unstructured", c.allowUnstructured, `suppress error reports when unstructured logging pattern is used`)

	return &analysis.Analyzer{
		Name:  "logcheck",
		Doc:   "Tool to check use of unstructured logging patterns.",
		Run: func(pass *analysis.Pass) (interface{}, error) {
			return run(pass, &c)
		},
		Flags: *logcheckFlags,
	}
}

func run(pass *analysis.Pass, c *config) (interface{}, error) {
```

Reply:
This is neat. Made the changes.